download/upgradeRecipe/upgradeRecipe7.0.adoc

```diff
@@ -1841,3 +1841,52 @@ public class MachineReassignmentIncrementalScoreCalculator
     }
 }
 ----
+
+
+[.upgrade-recipe-minor]
+=== Custom `MoveListFactory` and `MoveIteratorFactory`: `Solution_` parameter added to `Move` collections returned by factory methods.
+
+The `Solution_` type parameter was added to `Move` interface in 7.0.0.Beta6.
+The `MoveListFactory` and `MoveIteratorFactory` interfaces were subsequently updated so that you are not forced to
+use the raw-type `Move` in your custom `MoveListFactory` or `MoveIteratorFactory` implementation.
+
+Before in `*.java`:
+[source, java]
+----
+public class RowChangeMoveFactory implements MoveListFactory<NQueens> {
```

Comment:
Changing this code to use the cloud balancing example, as that's now the primary example (used in quick start) which most users are familiar with.